plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/SchemaInitializer.java

```diff
@@ -0,0 +1,102 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static java.util.Objects.requireNonNull;
+
+public class SchemaInitializer
+        implements Consumer<QueryRunner>
+{
+    private final String schemaName;
+    private final Map<String, String> schemaProperties;
+    private final Set<TpchTable<?>> clonedTpchTables;
+
+    private SchemaInitializer(String schemaName, Map<String, String> schemaProperties, Set<TpchTable<?>> tpchTablesToClone)
+    {
+        this.schemaName = requireNonNull(schemaName, "schemaName is null");
+        this.schemaProperties = requireNonNull(schemaProperties, "schemaProperties is null");
+        this.clonedTpchTables = requireNonNull(tpchTablesToClone, "tpchTablesToClone is null");
+    }
+
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @Override
+    public void accept(QueryRunner queryRunner)
+    {
+        String schemaProperties = this.schemaProperties.entrySet().stream()
+                .map(entry -> entry.getKey() + " = " + entry.getValue())
+                .collect(Collectors.joining(", ", " WITH ( ", " )"));
+        queryRunner.execute("CREATE SCHEMA IF NOT EXISTS " + schemaName + (this.schemaProperties.size() > 0 ? schemaProperties : ""));
+        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), clonedTpchTables);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private String schemaName = "tpch";
+        private Map<String, String> schemaProperties = ImmutableMap.of();
+        private Set<TpchTable<?>> cloneTpchTables = ImmutableSet.of();
+
+        public Builder withSchemaName(String schemaName)
+        {
+            this.schemaName = requireNonNull(schemaName, "schemaName is null");
+            return this;
+        }
+
+        public Builder withSchemaProperties(Map<String, String> schemaProperties)
+        {
+            this.schemaProperties = requireNonNull(schemaProperties, "schemaProperties is null");
+            return this;
+        }
+
+        public Builder withClonedTpchTables(TpchTable<?>... tpchTables)
+        {
+            this.cloneTpchTables = ImmutableSet.copyOf(tpchTables);
+            return this;
+        }
+
+        public Builder withClonedTpchTables(Iterable<TpchTable<?>> tpchTables, TpchTable<?>... additionalTpchTables)
```

Comment:
That's a rather odd overload. Let's just take `Iterable<TpchTable<?>> tpchTables`.